framework/include/base/Assembly.h

```diff
@@ -737,6 +737,10 @@ class Assembly
   /// Temporary work vector to keep from reallocating it
   std::vector<dof_id_type> _temp_dof_indices;
 
+  /// Temporary work data for reinitAtPhysical()
+  std::vector<Point> _temp_reference_points;
+  FEType _temp_fe_type;
```

Comment:
Coud you use one of the FE helpers instead? That way we do not need another FE and we use the helper in `reinitAtPhysical`. Unless I missed something else...